File: src/stores/zentao.js
import { defineStore } from 'pinia'
import { invoke } from '@tauri-apps/api/core'

export const useZentaoStore = defineStore('zentao', {
  state: () => ({
    token: null,
    projects: [],
    selectedProject: null,
    taskResults: [],
    creating: false,
    connected: false,
  }),

  getters: {
    successCount: (state) => state.taskResults.filter(r => r.task_created).length,
    failCount: (state) => state.taskResults.filter(r => !r.task_created).length,
  },

  actions: {
    async login(url, account, password) {
      try {
        this.token = await invoke('zentao_login', { url, account, password })
        this.connected = true
        return this.token
      } catch (e) {
        this.connected = false
        throw e
      }
    },

    async fetchProjects(url) {
      try {
        this.projects = await invoke('zentao_get_projects', {
          url,
          token: this.token,
        })
        return this.projects
      } catch (e) {
        console.error('获取项目列表失败:', e)
        throw e
      }
    },

    selectProject(project) {
      this.selectedProject = project
    },

    async createTask(url, branch, commits, projectId, assignedTo, taskType) {
      try {
        const result = await invoke('zentao_create_task', {
          url,
          token: this.token,
          branch,
          commits,
          projectId,
          assignedTo,
          taskType,
        })
        return result
      } catch (e) {
        console.error('创建任务失败:', e)
        throw e
      }
    },

    setResults(results) {
      this.taskResults = results
    },
  },
})

File: src-tauri/src/commands/zentao.rs
use tauri::command;
use crate::models::{ZentaoProject, TaskCreateResult, Commit};
use crate::services::{login, get_projects, create_task, build_task_payload};

#[command]
pub async fn zentao_login(url: String, account: String, password: String) -> Result<String, String> {
    login(&url, &account, &password).await
}

#[command]
pub async fn zentao_get_projects(url: String, token: String) -> Result<Vec<ZentaoProject>, String> {
    get_projects(&url, &token).await
}

#[command]
pub async fn zentao_create_task(
    url: String,
    token: String,
    branch: String,
    commits: Vec<Commit>,
    project_id: u32,
    assigned_to: String,
    task_type: String,
) -> Result<TaskCreateResult, String> {
    let task_data = build_task_payload(&branch, &commits, project_id, &assigned_to, &task_type);
    create_task(&url, &token, &task_data).await
}
